dialer_test.go

```diff
@@ -1016,3 +1016,64 @@ func TestDialerInitializesLazyCache(t *testing.T) {
 		t.Fatalf("dialer was initialized with non-lazy type: %T", tt)
 	}
 }
+
+type fakeResolver struct{}
+
+func (r *fakeResolver) Resolve(_ context.Context, name string) (instance.ConnName, error) {
+	// For TestDialerSuccessfullyDialsDnsTxtRecord
+	if name == "db.example.com" {
```

Comment:
Instead of relying on hardcoded values for particular tests, you could set fields in `fakeResolver` which makes this more future proof and easy to extend:

``` golang

type fakeResolve struct {
  instances map[string]struct{
    connName instance.ConnName
    err error
  }
}
```

And then this implementation becomes:

``` golang
func (r *fakeResolver) Resolve(_ context.Context, name string) (instance.ConnName, error) {
  icn, ok := r.instances[name]
  if !ok {
    return instance.ConnName{}, errors.New("not found")
  }
  return icn, nil
}
```

Reply:
This seems excessive. After simplifying obsolete test cases, `fakeResolver` only ever needs to resolve one domain name.

Reply:
I think it helps test readability. Otherwise, a person has to go searching for what's special about `db.example.com`.

If we're not supporting multiple instances anymore this also works:

``` golang
WithResolver(&fakeResolver{validName: "db.example.com"}),
```

``` golang
type fakeResolve struct {
  validName string
}

func (r *fakeResolver) Resolve(_ context.Context, name string) (instance.ConnName, error) {
  if name != r.validName {
    // return error
  }
  // return connection name
}
```

The point is to make the fake extensible -- not just for this use, but for future use.

Reply:
Actually this is what you do below in the resolver test -- let's do the same thing here.

Reply:
Done.